File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/rsp/RspRetailerOrders.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.rsp

import com.microblink.core.ScanResults
import org.json.JSONObject

class RspRetailerOrders(
    private val retailer: String,
    private val username: String,
    private val scan: ScanResults
): Rsp {
    override fun toJson(): JSONObject =
        JSONObject()
            .put("username", username)
            .put("retailer", retailer)
            .put("scan",  RspScan(scan).toJson())
}

File: ios/Plugin/Rsp/RspRetailerOrders.swift
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import BlinkReceipt
import BlinkEReceipt
import Capacitor

struct RspRetailerOrders : Rsp {
    private let provider: String
    private let username: String
    private let scan: BRScanResults
    
    func toJson() -> JSObject {
        JSObject.updateValue("username", username)
        JSObject.updateValue("provider", provider)
        JSObject.updateValue("scan",  RspScan(scan).toJson())
    }
}
